### ISSUE TITLE
A precision error appears on Conv2D's bias

### ISSUE THREAD

**Issue by Embed-Debuger** [pytorch/pytorch#92616]
### 🐛 Describe the bug

We found that torch.nn.Conv2d has a precision error. The specific reason comes from its bias parameter.
The reproduction of the error is specifically divided into four steps:
1. Initialize a Convolution operator (i.e. source_model), randomly input an input (i.e. data), and get an output (i.e. source_result);
2. Set a constant delta, set it to a large value, such as 10. Add the bias of source_model to delta to get a new Convolution operator (i.e. follow_model);
3. Input data into follow_model to get follow_result;
4. Calculate the distance between (source_result+delta) and follow_result.
Theoretically it should be small or even 0, but in practice a large result can be obtained. Moreover, the error distance of CPU is about ten times that of GPU. We also use the same method to run on tensorflow1.x and 2.x, and the calculated distance is 0, whether it is GPU or CPU.

```python
import numpy as np
import torch
import copy
from torch.nn import Conv2d
import os

def SourceModel(shape, device):
    model = Conv2d(in_channels=shape, out_channels=32, kernel_size=(5, 5)).to(device)
    return model

def FollowModel_4(source_model):
    # copy model
    follow_model = copy.deepcopy(source_model)
    # mutation model
    follow_model.train()
    follow_model.bias.data += delta
    follow_model.eval()
    return follow_model

DEVICE = "gpu"
if DEVICE == "gpu":
    # os.environ['TF_FORCE_GPU_ALLOW_GROWTH'] = 'true'
    os.environ["CUDA_DEVICE_ORDER"] = "PCI_BUS_ID"
    os.environ["CUDA_VISIBLE_DEVICES"] = "0"

device = torch.device('cuda' if DEVICE == "gpu" else "cpu")
delta = 10
shape = (10, 3, 32, 32)
data = torch.from_numpy(np.random.uniform(-1, 1, shape)).float().to(device)

source_model = SourceModel(shape[1], device).to(device)
follow_model = FollowModel_4(source_model)
source_result = source_model.forward(data) + delta
follow_result = follow_model.forward(data)
dis = torch.sum(torch.abs_(source_result-follow_result)).cpu().detach().numpy()[()]

print("device:", DEVICE, "; delta:", delta, "; dis:", dis)
```

```
device: gpu ; delta: 10 ; dis: 0.056635857
device: cpu ; delta: 10 ; dis: 0.4786234
```

### Versions

Collecting environment information...
PyTorch version: 1.10.0+cu102
Is debug build: False
CUDA used to build PyTorch: 10.2
ROCM used to build PyTorch: N/A

OS: Ubuntu 18.04.6 LTS (x86_64)
GCC version: (Ubuntu 7.5.0-3ubuntu1~18.04) 7.5.0
Clang version: Could not collect
CMake version: Could not collect
Libc version: glibc-2.17

Python version: 3.7.15 (default, Nov 24 2022, 21:12:53)  [GCC 11.2.0] (64-bit runtime)
Python platform: Linux-4.15.0-202-generic-x86_64-with-debian-buster-sid
Is CUDA available: True
CUDA runtime version: 10.1.105
CUDA_MODULE_LOADING set to: 
GPU models and configuration: GPU 0: GeForce RTX 2080 Ti
Nvidia driver version: 440.100
cuDNN version: Could not collect
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

Versions of relevant libraries:
[pip3] numpy==1.21.6
[pip3] torch==1.10.0+cu102
[pip3] torchaudio==0.10.0+rocm4.1
[pip3] torchvision==0.11.0+cu102
[conda] mkl                       2022.2.1                 pypi_0    pypi
[conda] mkl-fft                   1.3.1                    pypi_0    pypi
[conda] mkl-service               2.4.0                    pypi_0    pypi
[conda] numpy                     1.21.6                   pypi_0    pypi
[conda] torch                     1.10.0+cu102             pypi_0    pypi
[conda] torchaudio                0.10.0+rocm4.1           pypi_0    pypi
[conda] torchvision               0.11.0+cu102             pypi_0    pypi

cc @albanD @mruberry @jbschlosser @walterddr @saketh-are

**Comment by soulitzer** [comment]
Doing `print(torch.max(torch.abs_(source_result-follow_result)))` on cpu returns `tensor(1.2398e-05, grad_fn=<MaxBackward1>)`

**Comment by Embed-Debuger** [comment]
Thank you for your reply! I used your code to run on gpu and cpu respectively, and got 10 times different results. They should all be float32. I want to know why the difference exists.
```
device: cpu ; delta: 10 ; dis: 1.1444092e-05
device: gpu ; delta: 10 ; dis: 9.536743e-07
```

**Comment by albanD** [comment]
Hi,

You got slightly lucky on the GPU and a bit unlucky on the CPU for these particular values. But both are within expected range.
More details in https://pytorch.org/docs/stable/notes/numerical_accuracy.html

**Comment by jbschlosser** [comment]
Closing as resolved, but please feel free to reopen if you think there are issues beyond standard numerical accuracy details as outlined in @albanD's post.
